src/speed_challenge_ctrl.cpp

```diff
@@ -214,33 +235,50 @@ class SpeedChallenge{
         case states::MOVE_TO_POINT3:
 
             // Get point 3 
-
-            task_goal_pos_.point.x = point3_direction_vec_.x + starting_pos_.pose.position.y;
-            task_goal_pos_.point.y = point3_direction_vec_.y + starting_pos_.pose.position.x * -1;
+            task_goal_pos_.point.x = point3_direction_vec_.x + starting_pos_.pose.position.x;
+            task_goal_pos_.point.y = point3_direction_vec_.y + starting_pos_.pose.position.y;
             task_goal_pos_.task.current_task = task_master::Task::SPEED_RUN;
 
             task_pos_.publish(task_goal_pos_);
 
             if(isReached()){
 
-                status = states::RETURN_TO_START;
+                status = states::MOVE_BACK_TO_GATE;
 
                 ROS_INFO("Point 3 reached");
-                ROS_INFO("Moving back to start");
+                ROS_INFO("Moving back to gate");
             }
             break;
 
+        case states::MOVE_BACK_TO_GATE:
+
+            task_goal_pos_.point.x = gate_mid_point_.x;
+            task_goal_pos_.point.y = gate_mid_point_.y;
+
+            task_goal_pos_.task.current_task = task_master::Task::SPEED_RUN;
+            
+            task_pos_.publish(task_goal_pos_);
+
+            if(isReached()){
+
+                status = states::RETURN_TO_START;
+
+                ROS_INFO("Gate reached");
+                ROS_INFO("Returning to start");
+             }
+             break;
+
         case states::RETURN_TO_START:
 
-            task_goal_pos_.point.x = starting_pos_.pose.position.y;
-            task_goal_pos_.point.y = starting_pos_.pose.position.x * -1;
+            task_goal_pos_.point.x = starting_pos_.pose.position.x;
+            task_goal_pos_.point.y = starting_pos_.pose.position.y;
             task_goal_pos_.task.current_task = task_master::Task::SPEED_RUN;
 
             task_pos_.publish(task_goal_pos_);
 
             break;
        }
-    }
+}
 
     void globalPositionCallback(const geometry_msgs::PoseStamped::ConstPtr& msg) {
```

Comment:
that's actually a local position not global! so it could be renamed localPositionCallback or localPoseCallback. Pose includes position and orientation.